src/chains/ethereum/ethereum/tests/api/eth/feeHistory.test.ts

```diff
@@ -0,0 +1,525 @@
+
+            assert.strictEqual(feeHistory.oldestBlock, newestBlock);
+          });
+        });
+      });
+      describe("gasUsedRatio", () => {
+        it("returns 0 gasUsed for empty blocks", async () => {
+          const blockCount = "0x1";
+          const newestBlock = "0x2";
+          const feeHistory = await provider.send("eth_feeHistory", [
+            blockCount,
+            newestBlock,
+            []
+          ]);
+
+          assert.strictEqual(feeHistory.gasUsedRatio[0], 0);
+        });
+        it("returns gasUsed / maxGas", async () => {
+          await sendTransaction({ provider, to, from });
+
+          const block = await provider.send("eth_getBlockByNumber", ["latest"]);
+
+          const blockCount = "0x1";
+          const newestBlock = block.number;
+
+          const feeHistory = await provider.send("eth_feeHistory", [
+            blockCount,
+            newestBlock,
+            []
+          ]);
+
+          assert.strictEqual(feeHistory.gasUsedRatio.length, 1);
+          assert.strictEqual(
+            feeHistory.gasUsedRatio[0],
+            Number(block.gasUsed) / Number(block.gasLimit)
+          );
+        });
+        it("returns one entry for each block", async () => {
+          await mineNBlocks({ provider, blocks: 1 });
+          await sendTransaction({ provider, to, from });
+          await mineNBlocks({ provider, blocks: 1 });
+          const block = await provider.send("eth_getBlockByNumber", ["latest"]);
+
+          const blockCount = "0x2";
+          const newestBlock = block.number;
+
+          const feeHistory = await provider.send("eth_feeHistory", [
+            blockCount,
+            newestBlock,
+            []
+          ]);
+
+          assert.strictEqual(feeHistory.gasUsedRatio.length, 2);
+          assert.strictEqual(feeHistory.gasUsedRatio[0], 0);
+          assert.strictEqual(
+            feeHistory.gasUsedRatio[1],
+            Number(block.gasUsed) / Number(block.gasLimit)
+          );
+        });
+      });
+      describe("baseFeePerGas", () => {
+        it("returns blockCount + 1 baseFeePerGas", async () => {
+          const blockCount = "0x5";
+          const blocks = 5;
+          const newestBlock = "latest";
+          await mineNBlocks({ provider, blocks });
+
+          const feeHistory = await provider.send("eth_feeHistory", [
+            blockCount,
+            newestBlock,
+            []
+          ]);
+
+          assert.strictEqual(feeHistory.baseFeePerGas.length, blocks + 1);
+        });
+        it("baseFeePerGas matches block baseFeePerGas", async () => {
+          const blockCount = "0x5";
+          const blocks = 5;
+          const newestBlock = "latest";
+          await mineNBlocks({ provider, blocks });
+
+          const feeHistory = await provider.send("eth_feeHistory", [
+            blockCount,
+            newestBlock,
+            []
+          ]);
+
+          const block = await provider.send("eth_getBlockByNumber", [
```

Comment:
Maybe overkill, your call, but should this be done in a loop to verify _each_ base fee per gas matches its corresponding block?

Reply:
Yeah that's a fair one. 

My thinking is that since the fee is a function of the previous blocks' fees I think I am comfortable leaving the comparisons out. If the math is borked the tests in `ethereum-block` + the test for the next block should catch it. If the return order is incorrect, this test will fail.